src/video_core/amdgpu/liverpool.h

```diff
@@ -1277,6 +1279,7 @@ struct Liverpool {
     };
 
     Regs regs{};
+    std::array<ComputeProgram, NumComputeRings> asc_sh_regs{};
```

Comment:
An idea would be to store compute programs in mapped_queues and always access here with qid, so you don't need separate array and maintain a pointer